### PR TITLE
Fixes #12380 Refactored http requests into one function "httpRequest" in Util.php

### LINKED ISSUE #12380: One code to handle HTTP requests
Currently we have at least three places which handle HTTP requests and those are inconsistent. 

* Config::checkHTTP
* VersionInformation::getLatestVersion
* PMA_sendErrorReport

This should be all moved to single place and share as much code as possible.

### PR DESCRIPTION
**Pull request by Seldar**
Changed below methods to call the same method for httpRequests, which resides in Util class.

Util::httpRequest - new method
Config::checkHTTP - removed
VersionInformation::getLatestVersion - updated
PMA_sendErrorReport - updated

Fixed a couple of issues according to comments in #12457.

Signed-off-by: Volkan Ulukut arthan@gmail.com

### CODE REVIEW

File: test/classes/ConfigTest.php [hunk 1]
```diff
@@ -936,32 +936,4 @@ public function testIsGitRevision()
...
-        );
-        $this->assertFalse(
-            $this->object->checkHTTP("https://www.phpmyadmin.net/test/nothing")
-        );
-        // Use rate limit API as it's not subject to rate limiting
-        $this->assertContains(
-            '"resources"',
-            $this->object->checkHTTP("https://api.github.com/rate_limit", true)
-        );
-    }
```
Comment (nijel) [inline test/classes/ConfigTest.php]: Can you please adjust these tests to work on the new code? It really should be covered by tests.

**Comment by ibennetch** [comment]
Thanks for submitting a pull request. Could you please use your full/legal name for the signed-off-by?

File: test/classes/ConfigTest.php (hunk 1 above)
Comment (Seldar) [inline test/classes/ConfigTest.php]: Added new test cases.

**Comment by nijel** [comment]
Thanks, I've merged your changes, however some more fixes were necessary:

* 5a8666c Fix allow_url_fopen request handling
* 4382446 Consistent coding style
* d52e2cf Move Util::httpRequest tests to UtilTest.php

**Pull request merged**